domains/runtime/auto-id/src/lib.rs

```diff
@@ -0,0 +1,885 @@
+};
+use sp_messenger_host_functions::{get_storage_key, StorageKeyRequest};
+use sp_mmr_primitives::{EncodableOpaqueLeaf, Proof};
+use sp_runtime::generic::Era;
+use sp_runtime::traits::{
+    AccountIdLookup, BlakeTwo256, Block as BlockT, Checkable, Keccak256, One, SignedExtension,
+    ValidateUnsigned, Zero,
+};
+use sp_runtime::transaction_validity::{
+    InvalidTransaction, TransactionSource, TransactionValidity, TransactionValidityError,
+};
+use sp_runtime::{create_runtime_str, generic, impl_opaque_keys, ApplyExtrinsicResult, Digest};
+pub use sp_runtime::{MultiAddress, Perbill, Permill};
+use sp_std::marker::PhantomData;
+use sp_std::prelude::*;
+use sp_subspace_mmr::domain_mmr_runtime_interface::verify_mmr_proof;
+use sp_subspace_mmr::MmrLeaf;
+use sp_version::RuntimeVersion;
+use subspace_runtime_primitives::{
+    BlockNumber as ConsensusBlockNumber, Hash as ConsensusBlockHash, Moment,
+    SlowAdjustingFeeUpdate, SSC,
+};
+
+/// Block type as expected by this runtime.
+pub type Block = generic::Block<Header, UncheckedExtrinsic>;
+
+/// A Block signed with a Justification
+pub type SignedBlock = generic::SignedBlock<Block>;
+
+/// BlockId type as expected by this runtime.
+pub type BlockId = generic::BlockId<Block>;
+
+/// The SignedExtension to the basic transaction logic.
+pub type SignedExtra = (
+    frame_system::CheckNonZeroSender<Runtime>,
+    frame_system::CheckSpecVersion<Runtime>,
+    frame_system::CheckTxVersion<Runtime>,
+    frame_system::CheckGenesis<Runtime>,
+    frame_system::CheckMortality<Runtime>,
+    frame_system::CheckNonce<Runtime>,
+    frame_system::CheckWeight<Runtime>,
+    pallet_transaction_payment::ChargeTransactionPayment<Runtime>,
+);
+
+/// Unchecked extrinsic type as expected by this runtime.
+pub type UncheckedExtrinsic =
+    generic::UncheckedExtrinsic<Address, RuntimeCall, Signature, SignedExtra>;
+
+/// Extrinsic type that has already been checked.
+pub type CheckedExtrinsic = generic::CheckedExtrinsic<AccountId, RuntimeCall, SignedExtra>;
+
+/// Executive: handles dispatch to the various modules.
+pub type Executive = domain_pallet_executive::Executive<
+    Runtime,
+    frame_system::ChainContext<Runtime>,
+    Runtime,
+    AllPalletsWithSystem,
+>;
+
+impl_opaque_keys! {
+    pub struct SessionKeys {
+        /// Primarily used for adding the operator signing key into the Keystore.
+        pub operator: sp_domains::OperatorKey,
+    }
+}
+
+#[sp_version::runtime_version]
+pub const VERSION: RuntimeVersion = RuntimeVersion {
+    spec_name: create_runtime_str!("subspace-auto-id-domain"),
+    impl_name: create_runtime_str!("subspace-auto-id-domain"),
+    authoring_version: 0,
+    spec_version: 0,
+    impl_version: 0,
+    apis: RUNTIME_API_VERSIONS,
+    transaction_version: 0,
+    state_version: 0,
+    extrinsic_state_version: 1,
+};
+
+parameter_types! {
+    pub const Version: RuntimeVersion = VERSION;
+    pub const BlockHashCount: BlockNumber = 2400;
```

Comment:
Not related to this PR, it seems consensus runtime also has this const value set to 2400 but it is never used, `frame_system::Config` uses `ConstU32<250>` instead. cc @nazar-pc

Reply:
Yeah, that seems to have been in the code from the very beginning, we should fix it

Reply:
I have updated this to 250 but current Gemini runtimes will continue to hold the old hashes due to the way removal is handled in the frame-system. New networks will be as usual.
@nazar-pc let me know if this is undesirable, i can reset it back with a todo for removal at a later time

Reply:
No need, this `BlockHashCount` wasn't actually used, it was hardcoded to `ConstU32<250>` below. You can either replace `ConstU32<250>` or remove `BlockHashCount`.